openfasoc/generators/glayout/test_glayout.py

```diff
@@ -0,0 +1,157 @@
+        package_name = parts[0]
+        if len(parts) == 2:
+            required_version = parts[1]
+            print(f"\n...Checking {package_name}=={parts[1]}...")
+        else:
+            required_version = None  # No specific version required
+            print(f"\n...Checking {package_name}...")
+        try:
+            installed_version = metadata.version(package_name)
+            if required_version and installed_version != required_version:
+                raise ImportError(f"Version mismatch for {package_name}: Expected {required_version}, found {installed_version}. Consider running 'pip install -r {requirements_file}'")
+        except metadata.PackageNotFoundError:
+            raise ImportError(f"Requirement {package_name} not installed. Consider running 'pip install -r {requirements_file}'")
+        print(f"\n{package_name} is installed!")
+        
+# function to check for system-wide tools like Magic, Netgen, and Ngspice
+def check_system_tools(tool_names, miniconda3_path):
+    print_heading("Checking system tools")
+    for tool in tool_names:
+        try:
+            print(f"\n...Checking {tool}...")
+            result = subprocess.run([tool, '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, check=True)
+            print(f"\n{tool} is installed!")
+        except subprocess.CalledProcessError as e:
+            raise EnvironmentError(f"{tool} not installed or not in PATH. Please install it.") from e
+    print("\n...Checking netgen...")
+    netgen_path = miniconda3_path / "lib" / "netgen"
+    # print(str(netgen_path))
+    if not netgen_path.exists():
+        raise EnvironmentError("Netgen not found in expected location!")
+    print("\nnetgen is installed!")
+# List of system tools to check
+
+def check_miniconda3_and_pdk():
+    print_heading("Checking Miniconda3 and PDK")
+    # Check if miniconda3 is installed
+    try:
+        result = subprocess.run(['conda', '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, check=True)
+        print(f"\nMiniconda3 is installed: {result.stdout}")
+    except subprocess.CalledProcessError as e:
+        raise EnvironmentError("Miniconda3 not installed or not in PATH. Please install it.") from e
+
+    paths_to_check = [
+        Path("/usr/bin/miniconda3/share/pdk/"), 
+        Path(f"/home/{os.getenv('LOGNAME')}/miniconda3/share/pdk/")
+    ]
+    pdk_root = None
+    for path in paths_to_check:
+        print(f"\n...Checking for PDK root at: {path}...")
+        if path.exists():
+            pdk_root = path
+            break
+
+    if not pdk_root:
+        raise EnvironmentError("PDK root not found in expected locations!")
+    else:
+        print(f"\nPDK root found at: {pdk_root}")
+    
+    miniconda3_path = pdk_root.resolve().parents[1]
+    required_pdk_dirs = ["sky130A", "gf180mcuC"]
+    missing_dirs = [pdk_dir for pdk_dir in required_pdk_dirs if not (pdk_root / pdk_dir).exists()]
```

Comment:
@chetanyagoyal can you make it so that if one PDK is found it just does a warning. 
also maybe rename "required_pdks" to "supported_pdks" or something
Along this lines of:
```
import warnings
if len(missing_dirs)==len(required_pdks):
	raise .... whatever error blah blah
elif len(missing_dirs) < len(required_pdks): 
	warnings.warn("you dont have all the supported PDKs installed")
# else do nothing

Reply:
Also, you may want to check if a tool is installed by trying to run it and look for return code.

Reply:
will do, i will raise a warning instead for the pdks

Reply:
addressed